### PR TITLE
testifylint: enable go-require, float-compare and require-error rules

### PR DESCRIPTION
**Pull request by mmorel-35**
Thank you for contributing to Velero!

# Please add a summary of your change

Enable and fixes go-require and require-error rules of testifylint linter.

Also update golangci-lint version

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by kaovilai — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -79,9 +80,9 @@ func TestConfig(t *testing.T) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			client, err := Config(test.kubeconfig, test.kubecontext, "velero", test.QPS, test.burst)
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			assert.Equal(t, test.expectedHost, client.Host)
-			assert.Equal(t, test.QPS, client.QPS)
+			assert.InDelta(t, test.QPS, client.QPS, 0.01)
```
Comment (blackpiglet) [inline pkg/client/client_test.go:85]: Why do we need to allow a difference between the expected and actual result here?

File: pkg/client/factory_test.go [hunk 2]
```diff
@@ -108,7 +109,7 @@ func TestFactory(t *testing.T) {
 			flags.Parse([]string{"--kubeconfig", test.kubeconfig, "--kubecontext", test.kubecontext})
 			clientConfig, _ := f.ClientConfig()
 			assert.Equal(t, test.expectedHost, clientConfig.Host)
-			assert.Equal(t, test.QPS, clientConfig.QPS)
+			assert.InDelta(t, test.QPS, clientConfig.QPS, 0.01)
```
Comment (blackpiglet) [inline pkg/client/factory_test.go:112]: Also the same question as above.

File: pkg/cmd/cli/backup/logs_test.go [hunk 3]
```diff
@@ -136,7 +136,7 @@ func TestNewLogsCommand(t *testing.T) {
 		done := make(chan bool)
 		go func() {
 			err = l.Run(c, f)
-			require.Error(t, err)
+			assert.Error(t, err)
```
Comment (blackpiglet) [inline pkg/cmd/cli/backup/logs_test.go:139]: Why do we need to change from require to assert here?

File: pkg/restore/restore_test.go [hunk 4]
```diff
@@ -2340,21 +2339,15 @@ func assertRestoredItems(t *testing.T, h *harness, want []*test.APIResource) {
 			}
 
 			res, err := client.Get(context.TODO(), item.GetName(), metav1.GetOptions{})
-			if !assert.NoError(t, err) {
-				continue
-			}
+			require.NoError(t, err)
```
Comment (blackpiglet) [inline pkg/restore/restore_test.go:2342]: I'm a little confused here.
Does the original code want the testing flow going on even meeting an error?
If so, it seems the change here seems not right.

File: pkg/util/kube/pvc_pv_test.go [hunk 5]
```diff
@@ -125,9 +126,9 @@ func TestWaitPVCBound(t *testing.T) {
 			pv, err := WaitPVCBound(context.Background(), kubeClient.CoreV1(), kubeClient.CoreV1(), test.pvcName, test.pvcNamespace, time.Millisecond)
 
 			if err != nil {
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply pkg/util/kube/pvc_pv_test.go:128]: Thanks. I agree with you. Although both should work, the code should check the `test.err` instead.

**Review by blackpiglet — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

File: internal/hook/item_hook_handler_test.go [hunk 6]
```diff
@@ -490,7 +490,7 @@ func TestHandleHooks(t *testing.T) {
 			err := h.HandleHooks(velerotest.NewLogger(), groupResource, test.item, test.hooks, test.phase, hookTracker)
 
 			if test.expectedError != nil {
-				assert.EqualError(t, err, test.expectedError.Error())
+				require.EqualError(t, err, test.expectedError.Error())
 				return
```
Comment (mateusoliveira43) [inline internal/hook/item_hook_handler_test.go:494]: is this return needed?

File: internal/hook/item_hook_handler_test.go [hunk 7]
```diff
@@ -1381,7 +1381,11 @@ func TestGetRestoreHooksFromSpec(t *testing.T) {
 			actual, err := GetRestoreHooksFromSpec(tc.hookSpec)
 
 			assert.Equal(t, tc.expected, actual)
-			assert.Equal(t, tc.expectedError, err)
+			if tc.expectedError != nil {
+				require.EqualError(t, err, tc.expectedError.Error())
+				return
```
Comment (mateusoliveira43) [inline internal/hook/item_hook_handler_test.go:1386]: is this return needed?

File: pkg/backup/actions/csi/pvc_action_test.go [hunk 8]
```diff
@@ -175,29 +176,29 @@ func TestExecute(t *testing.T) {
 					var vsList v1.VolumeSnapshotList
 					err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 10*time.Second, true, func(ctx context.Context) (bool, error) {
 						err = pvcBIA.crClient.List(ctx, &vsList, &crclient.ListOptions{Namespace: tc.pvc.Namespace})
-
-						require.NoError(t, err)
+						//nolint:testifylint // false-positive
```
Comment (mateusoliveira43) [inline pkg/backup/actions/csi/pvc_action_test.go:179]: this line should not be needed

running locally, give me an error `directive `//nolint:testifylint // false-positive` is unused for linter "testifylint" (nolintlint)`

Comment (mateusoliveira43) [reply pkg/backup/actions/csi/pvc_action_test.go:179]: `golangci-lint has version v1.59.1 built with go1.22.3`

File: internal/hook/item_hook_handler_test.go [hunk 9]
```diff
@@ -490,11 +490,10 @@ func TestHandleHooks(t *testing.T) {
 			err := h.HandleHooks(velerotest.NewLogger(), groupResource, test.item, test.hooks, test.phase, hookTracker)
 
 			if test.expectedError != nil {
-				assert.EqualError(t, err, test.expectedError.Error())
-				return
+				require.EqualError(t, err, test.expectedError.Error())
+			} else {
+				require.NoError(t, err)
```
Comment (kaovilai) [inline internal/hook/item_hook_handler_test.go:495]: I feel like require.EqualError should be sufficient here without if/else as long as you don't call `.Error()`
Comparing error type to error type should be fine IMO.

Comment (kaovilai) [reply internal/hook/item_hook_handler_test.go:495]: ok.

**Pull request closed without merging**